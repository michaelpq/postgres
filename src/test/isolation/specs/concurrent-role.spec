# Test for interactions with DDL commands manipulating role
# objects.
# The following set of commands can interact in concurrency:
# - ALTER ROLE SET
# - ALTER ROLE option

setup
{
  CREATE ROLE regress_role_conc;
}

teardown
{
  DROP ROLE regress_role_conc;
}

session "s1"
step "s1_begin"			{ BEGIN; }
step "s1_alterrole_set"	{
	ALTER ROLE regress_role_conc SET commit_delay = '10'; }
step "s1_alterrole_opt"	{
	ALTER ROLE regress_role_conc PASSWORD 'foo'; }
step "s1_commit"	{ COMMIT; }

session "s2"
step "s2_begin"			{ BEGIN; }
step "s2_alterrole_set"	{
	ALTER ROLE regress_role_conc SET commit_delay = '10'; }
step "s2_alterrole_opt"	{
	ALTER ROLE regress_role_conc PASSWORD 'foo'; }
step "s2_commit"	{ COMMIT; }

permutation "s1_begin" "s1_alterrole_set" "s2_begin" "s2_alterrole_set"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_alterrole_set" "s2_begin" "s2_alterrole_opt"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_alterrole_opt" "s2_begin" "s2_alterrole_set"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_alterrole_opt" "s2_begin" "s2_alterrole_opt"
	"s1_commit" "s2_commit"
