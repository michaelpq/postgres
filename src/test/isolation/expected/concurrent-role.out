Parsed test spec with 2 sessions

starting permutation: s1_begin s1_alterrole_set s2_begin s2_alterrole_set s1_commit s2_commit
step s1_begin: BEGIN;
step s1_alterrole_set: 
	ALTER ROLE regress_role_conc SET commit_delay = '10';
step s2_begin: BEGIN;
step s2_alterrole_set: 
	ALTER ROLE regress_role_conc SET commit_delay = '10'; <waiting ...>
step s1_commit: COMMIT;
step s2_alterrole_set: <... completed>
error in steps s1_commit s2_alterrole_set: ERROR:  duplicate key value violates unique constraint "pg_db_role_setting_databaseid_rol_index"
step s2_commit: COMMIT;

starting permutation: s1_begin s1_alterrole_set s2_begin s2_alterrole_opt s1_commit s2_commit
step s1_begin: BEGIN;
step s1_alterrole_set: 
	ALTER ROLE regress_role_conc SET commit_delay = '10';
step s2_begin: BEGIN;
step s2_alterrole_opt: 
	ALTER ROLE regress_role_conc PASSWORD 'foo';
step s1_commit: COMMIT;
step s2_commit: COMMIT;

starting permutation: s1_begin s1_alterrole_opt s2_begin s2_alterrole_set s1_commit s2_commit
step s1_begin: BEGIN;
step s1_alterrole_opt: 
	ALTER ROLE regress_role_conc PASSWORD 'foo';
step s2_begin: BEGIN;
step s2_alterrole_set: 
	ALTER ROLE regress_role_conc SET commit_delay = '10';
step s1_commit: COMMIT;
step s2_commit: COMMIT;

starting permutation: s1_begin s1_alterrole_opt s2_begin s2_alterrole_opt s1_commit s2_commit
step s1_begin: BEGIN;
step s1_alterrole_opt: 
	ALTER ROLE regress_role_conc PASSWORD 'foo';
step s2_begin: BEGIN;
step s2_alterrole_opt: 
	ALTER ROLE regress_role_conc PASSWORD 'foo'; <waiting ...>
step s1_commit: COMMIT;
step s2_alterrole_opt: <... completed>
error in steps s1_commit s2_alterrole_opt: ERROR:  tuple concurrently updated
step s2_commit: COMMIT;
